### ISSUE TITLE
add visual indication when sentry truncates dict data

### ISSUE THREAD

**Issue by ghazi-git** [getsentry/sentry#68426]
### Problem Statement

My understanding is that sentry has a limit on the amount of data it collects, so it sometimes truncates the data to make sure it doesn't exceed that limit. This is especially clear for long strings (like queries in the breadcrumbs section) with the added ellipsis at the end, but not so much for python dicts.

Here's one example of what I see when there is an error when receiving a postmark webhook (and here's the [complete object for reference](https://postmarkapp.com/developer/webhooks/open-tracking-webhook#open-webhook-data)):
![image](https://user-images.githubusercontent.com/17159441/215505974-74ea8f05-e227-4cf5-90db-11f442d6891c.png)
In the above example, I'm sure sentry has truncated the data because I can see the postmark metadata in one frame of the stack trace (we assign the metadata to a separate variable with `metadata = payload["Metadata"]` and I can see the contents of that variable)

The problem is that the dict representation doesn't tell me if sentry has truncated the data or not. So, now for each error, I need to think twice: did sentry truncate the dict data or did the client of my code not submit the key I'm expecting to find.

### Solution Brainstorm

I think a simple ellipsis at the end of each truncated dict would be a good visual indication that sentry has truncated the data.
```
{
  "key1": "value1",
  "key2": "value2",
  ...
}
```

However, I'm not sure what happens to existing data especially if sentry didn't collect before information about if variables were truncated or not.

**Comment by MichaelSun48** [comment]
Hey there @ghazi-git! 👋 This is really good feedback - definitely agree that the issue details page should be transparent about which parts of the body we are truncating. I'm going to add this to the feature request backlog for now and surface it with the issue details team today. Thanks for writing in about this!

**Comment by sl0thentr0py** [comment]
another request
https://github.com/getsentry/sentry-python/issues/3262

**Comment by aberres** [comment]
I am also running into stripped dicts and lists.

Having an error message telling you about duplicates in a list without any visible duplicates is, well, unfortunate.

**Comment by tsx** [comment]
Hi there! At [Close](https://close.com) we recently migrated to Sentry from a competitor, and we love the product so far but this truncating issue is tripping us regularly. Our engineers are regularly getting confused about what data was in the input while debugging issues. At times it feels like it'd save us time to not send the argument/locals data to Sentry at all.

Are there any workarounds we could apply?

**Comment by raulparada** [comment]
Hi. A paying customer here that wants to share the same frustration already expressed on this thread after losing hours scratching my head because of this. Guys (PMs, devs at Sentry) seriously, your core product is issue monitoring, and if you modify the issue data without giving _any_ indication to users, then that's a major UX flaw. That's like a logging solution just modifying/dropping logs due to reasons that make total sense to the service-provider but no sense what-so-ever to the customer. This should not maybe be on a backlog for maybe three quarters down the line maybe. This is your core product!

**Comment by olejorgenb** [comment]
Agree, the way it's displayed now is extremely misleading. I understand that it have to be a limit, but it must be clearly visible when the data has been truncated. Otherwise the info has close to negative value.
